src/vendor/libinjection/src/libinjection_sqli.c

```diff
@@ -314,6 +302,12 @@ static void st_assign(stoken_t * st, const char stype,
     st->val[last] = CHAR_NULL;
 }
 
+static void sf_update_state(struct libinjection_sqli_state * sf, const char stype,
+                            size_t pos, size_t len, const char* value) {
+    st_assign(sf->current, stype, pos, len, value);
+    sf->hadWordBoundary = stype == TYPE_COMMENT;
```

Comment:
`st_assign` uses `memcpy` while `st_assign_char` only had assignments. I think that's why the microbenchmark increased quite a bit:

```
scenario:is_sqli_matcher.random
🟥 execution_time [+477.773µs; +478.102µs] or [+643.703%; +644.145%]
```